// @typescript-eslint/scope-manager/dist/lib/dom.js
"use strict";
// THIS CODE WAS AUTOMATICALLY GENERATED
// DO NOT EDIT THIS CODE BY HAND
// YOU CAN REGENERATE IT USING yarn generate:lib
Object.defineProperty(exports, "__esModule", { value: true });
exports.dom = void 0;
exports.dom = {
    Account: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Account',
    },
    AddEventListenerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AddEventListenerOptions',
    },
    AesCbcParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesCbcParams',
    },
    AesCtrParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesCtrParams',
    },
    AesDerivedKeyParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesDerivedKeyParams',
    },
    AesGcmParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesGcmParams',
    },
    AesKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesKeyAlgorithm',
    },
    AesKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesKeyGenParams',
    },
    Algorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Algorithm',
    },
    AnalyserOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnalyserOptions',
    },
    AnimationEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnimationEventInit',
    },
    AnimationPlaybackEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnimationPlaybackEventInit',
    },
    AssertionOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AssertionOptions',
    },
    AssignedNodesOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AssignedNodesOptions',
    },
    AudioBufferOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioBufferOptions',
    },
    AudioBufferSourceOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioBufferSourceOptions',
    },
    AudioContextInfo: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioContextInfo',
    },
    AudioContextOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioContextOptions',
    },
    AudioNodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioNodeOptions',
    },
    AudioParamDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioParamDescriptor',
    },
    AudioProcessingEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioProcessingEventInit',
    },
    AudioTimestamp: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioTimestamp',
    },
    AudioWorkletNodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioWorkletNodeOptions',
    },
    AuthenticationExtensionsClientInputs: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AuthenticationExtensionsClientInputs',
    },
    AuthenticationExtensionsClientOutputs: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AuthenticationExtensionsClientOutputs',
    },
    AuthenticatorSelectionCriteria: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AuthenticatorSelectionCriteria',
    },
    BiquadFilterOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BiquadFilterOptions',
    },
    BlobPropertyBag: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BlobPropertyBag',
    },
    ByteLengthChunk: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ByteLengthChunk',
    },
    CacheQueryOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CacheQueryOptions',
    },
    CanvasRenderingContext2DSettings: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasRenderingContext2DSettings',
    },
    ChannelMergerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ChannelMergerOptions',
    },
    ChannelSplitterOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ChannelSplitterOptions',
    },
    ClientData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ClientData',
    },
    ClientQueryOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ClientQueryOptions',
    },
    ClipboardEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ClipboardEventInit',
    },
    CloseEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CloseEventInit',
    },
    CompositionEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CompositionEventInit',
    },
    ComputedEffectTiming: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ComputedEffectTiming',
    },
    ComputedKeyframe: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ComputedKeyframe',
    },
    ConfirmSiteSpecificExceptionsInformation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConfirmSiteSpecificExceptionsInformation',
    },
    ConstantSourceOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConstantSourceOptions',
    },
    ConstrainBooleanParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConstrainBooleanParameters',
    },
    ConstrainDOMStringParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConstrainDOMStringParameters',
    },
    ConstrainDoubleRange: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConstrainDoubleRange',
    },
    ConstrainULongRange: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConstrainULongRange',
    },
    ConstrainVideoFacingModeParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConstrainVideoFacingModeParameters',
    },
    ConvolverOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConvolverOptions',
    },
    CredentialCreationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CredentialCreationOptions',
    },
    CredentialPropertiesOutput: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CredentialPropertiesOutput',
    },
    CredentialRequestOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CredentialRequestOptions',
    },
    CustomEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CustomEventInit',
    },
    DOMMatrix2DInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DOMMatrix2DInit',
    },
    DOMMatrixInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DOMMatrixInit',
    },
    DOMPointInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DOMPointInit',
    },
    DOMQuadInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DOMQuadInit',
    },
    DOMRectInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DOMRectInit',
    },
    DelayOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DelayOptions',
    },
    DeviceLightEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceLightEventInit',
    },
    DeviceMotionEventAccelerationInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceMotionEventAccelerationInit',
    },
    DeviceMotionEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceMotionEventInit',
    },
    DeviceMotionEventRotationRateInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceMotionEventRotationRateInit',
    },
    DeviceOrientationEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceOrientationEventInit',
    },
    DevicePermissionDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DevicePermissionDescriptor',
    },
    DocumentTimelineOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DocumentTimelineOptions',
    },
    DoubleRange: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DoubleRange',
    },
    DragEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DragEventInit',
    },
    DynamicsCompressorOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DynamicsCompressorOptions',
    },
    EcKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EcKeyAlgorithm',
    },
    EcKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EcKeyGenParams',
    },
    EcKeyImportParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EcKeyImportParams',
    },
    EcdhKeyDeriveParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EcdhKeyDeriveParams',
    },
    EcdsaParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EcdsaParams',
    },
    EffectTiming: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EffectTiming',
    },
    ElementCreationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ElementCreationOptions',
    },
    ElementDefinitionOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ElementDefinitionOptions',
    },
    ErrorEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ErrorEventInit',
    },
    EventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventInit',
    },
    EventListenerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventListenerOptions',
    },
    EventModifierInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventModifierInit',
    },
    EventSourceInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventSourceInit',
    },
    ExceptionInformation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ExceptionInformation',
    },
    FilePropertyBag: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FilePropertyBag',
    },
    FocusEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FocusEventInit',
    },
    FocusNavigationEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FocusNavigationEventInit',
    },
    FocusNavigationOrigin: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FocusNavigationOrigin',
    },
    FocusOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FocusOptions',
    },
    FullscreenOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FullscreenOptions',
    },
    GainOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GainOptions',
    },
    GamepadEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GamepadEventInit',
    },
    GetNotificationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GetNotificationOptions',
    },
    GetRootNodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GetRootNodeOptions',
    },
    HashChangeEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HashChangeEventInit',
    },
    HkdfParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HkdfParams',
    },
    HmacImportParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HmacImportParams',
    },
    HmacKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HmacKeyAlgorithm',
    },
    HmacKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HmacKeyGenParams',
    },
    IDBIndexParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'IDBIndexParameters',
    },
    IDBObjectStoreParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'IDBObjectStoreParameters',
    },
    IDBVersionChangeEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'IDBVersionChangeEventInit',
    },
    IIRFilterOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'IIRFilterOptions',
    },
    ImageBitmapOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ImageBitmapOptions',
    },
    ImageBitmapRenderingContextSettings: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ImageBitmapRenderingContextSettings',
    },
    ImageEncodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ImageEncodeOptions',
    },
    InputEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'InputEventInit',
    },
    IntersectionObserverEntryInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'IntersectionObserverEntryInit',
    },
    IntersectionObserverInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'IntersectionObserverInit',
    },
    JsonWebKey: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'JsonWebKey',
    },
    KeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'KeyAlgorithm',
    },
    KeyboardEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'KeyboardEventInit',
    },
    Keyframe: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Keyframe',
    },
    KeyframeAnimationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'KeyframeAnimationOptions',
    },
    KeyframeEffectOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'KeyframeEffectOptions',
    },
    MediaElementAudioSourceOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaElementAudioSourceOptions',
    },
    MediaEncryptedEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaEncryptedEventInit',
    },
    MediaKeyMessageEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaKeyMessageEventInit',
    },
    MediaKeySystemConfiguration: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaKeySystemConfiguration',
    },
    MediaKeySystemMediaCapability: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaKeySystemMediaCapability',
    },
    MediaQueryListEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaQueryListEventInit',
    },
    MediaStreamAudioSourceOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaStreamAudioSourceOptions',
    },
    MediaStreamConstraints: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaStreamConstraints',
    },
    MediaStreamErrorEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaStreamErrorEventInit',
    },
    MediaStreamEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaStreamEventInit',
    },
    MediaStreamTrackAudioSourceOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaStreamTrackAudioSourceOptions',
    },
    MediaStreamTrackEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaStreamTrackEventInit',
    },
    MediaTrackCapabilities: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaTrackCapabilities',
    },
    MediaTrackConstraintSet: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaTrackConstraintSet',
    },
    MediaTrackConstraints: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaTrackConstraints',
    },
    MediaTrackSettings: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaTrackSettings',
    },
    MediaTrackSupportedConstraints: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MediaTrackSupportedConstraints',
    },
    MessageEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MessageEventInit',
    },
    MidiPermissionDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MidiPermissionDescriptor',
    },
    MouseEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MouseEventInit',
    },
    MultiCacheQueryOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MultiCacheQueryOptions',
    },
    MutationObserverInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'MutationObserverInit',
    },
    NavigationPreloadState: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'NavigationPreloadState',
    },
    NotificationAction: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'NotificationAction',
    },
    NotificationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'NotificationOptions',
    },
    OfflineAudioCompletionEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'OfflineAudioCompletionEventInit',
    },
    OfflineAudioContextOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'OfflineAudioContextOptions',
    },
    OptionalEffectTiming: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'OptionalEffectTiming',
    },
    OscillatorOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'OscillatorOptions',
    },
    PageTransitionEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PageTransitionEventInit',
    },
    PannerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PannerOptions',
    },
    PaymentCurrencyAmount: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentCurrencyAmount',
    },
    PaymentDetailsBase: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentDetailsBase',
    },
    PaymentDetailsInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentDetailsInit',
    },
    PaymentDetailsModifier: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentDetailsModifier',
    },
    PaymentDetailsUpdate: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentDetailsUpdate',
    },
    PaymentItem: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentItem',
    },
    PaymentMethodData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentMethodData',
    },
    PaymentOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentOptions',
    },
    PaymentRequestUpdateEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentRequestUpdateEventInit',
    },
    PaymentShippingOption: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PaymentShippingOption',
    },
    Pbkdf2Params: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Pbkdf2Params',
    },
    PerformanceObserverInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PerformanceObserverInit',
    },
    PeriodicWaveConstraints: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PeriodicWaveConstraints',
    },
    PeriodicWaveOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PeriodicWaveOptions',
    },
    PermissionDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PermissionDescriptor',
    },
    PipeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PipeOptions',
    },
    PointerEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PointerEventInit',
    },
    PopStateEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PopStateEventInit',
    },
    PositionOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PositionOptions',
    },
    PostMessageOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PostMessageOptions',
    },
    ProgressEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ProgressEventInit',
    },
    PromiseRejectionEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PromiseRejectionEventInit',
    },
    PropertyIndexedKeyframes: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PropertyIndexedKeyframes',
    },
    PublicKeyCredentialCreationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialCreationOptions',
    },
    PublicKeyCredentialDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialDescriptor',
    },
    PublicKeyCredentialEntity: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialEntity',
    },
    PublicKeyCredentialParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialParameters',
    },
    PublicKeyCredentialRequestOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialRequestOptions',
    },
    PublicKeyCredentialRpEntity: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialRpEntity',
    },
    PublicKeyCredentialUserEntity: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PublicKeyCredentialUserEntity',
    },
    PushPermissionDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PushPermissionDescriptor',
    },
    PushSubscriptionJSON: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PushSubscriptionJSON',
    },
    PushSubscriptionOptionsInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'PushSubscriptionOptionsInit',
    },
    QueuingStrategy: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'QueuingStrategy',
    },
    RTCAnswerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCAnswerOptions',
    },
    RTCCertificateExpiration: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCCertificateExpiration',
    },
    RTCConfiguration: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCConfiguration',
    },
    RTCDTMFToneChangeEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCDTMFToneChangeEventInit',
    },
    RTCDataChannelEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCDataChannelEventInit',
    },
    RTCDataChannelInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCDataChannelInit',
    },
    RTCDtlsFingerprint: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCDtlsFingerprint',
    },
    RTCDtlsParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCDtlsParameters',
    },
    RTCErrorEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCErrorEventInit',
    },
    RTCErrorInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCErrorInit',
    },
    RTCIceCandidateAttributes: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceCandidateAttributes',
    },
    RTCIceCandidateComplete: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceCandidateComplete',
    },
    RTCIceCandidateDictionary: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceCandidateDictionary',
    },
    RTCIceCandidateInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceCandidateInit',
    },
    RTCIceCandidatePair: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceCandidatePair',
    },
    RTCIceCandidatePairStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceCandidatePairStats',
    },
    RTCIceGatherOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceGatherOptions',
    },
    RTCIceParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceParameters',
    },
    RTCIceServer: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIceServer',
    },
    RTCIdentityProviderOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCIdentityProviderOptions',
    },
    RTCInboundRTPStreamStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCInboundRTPStreamStats',
    },
    RTCMediaStreamTrackStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCMediaStreamTrackStats',
    },
    RTCOAuthCredential: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCOAuthCredential',
    },
    RTCOfferAnswerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCOfferAnswerOptions',
    },
    RTCOfferOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCOfferOptions',
    },
    RTCOutboundRTPStreamStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCOutboundRTPStreamStats',
    },
    RTCPeerConnectionIceErrorEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCPeerConnectionIceErrorEventInit',
    },
    RTCPeerConnectionIceEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCPeerConnectionIceEventInit',
    },
    RTCRTPStreamStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRTPStreamStats',
    },
    RTCRtcpFeedback: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtcpFeedback',
    },
    RTCRtcpParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtcpParameters',
    },
    RTCRtpCapabilities: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpCapabilities',
    },
    RTCRtpCodecCapability: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpCodecCapability',
    },
    RTCRtpCodecParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpCodecParameters',
    },
    RTCRtpCodingParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpCodingParameters',
    },
    RTCRtpContributingSource: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpContributingSource',
    },
    RTCRtpDecodingParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpDecodingParameters',
    },
    RTCRtpEncodingParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpEncodingParameters',
    },
    RTCRtpFecParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpFecParameters',
    },
    RTCRtpHeaderExtension: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpHeaderExtension',
    },
    RTCRtpHeaderExtensionCapability: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpHeaderExtensionCapability',
    },
    RTCRtpHeaderExtensionParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpHeaderExtensionParameters',
    },
    RTCRtpParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpParameters',
    },
    RTCRtpReceiveParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpReceiveParameters',
    },
    RTCRtpRtxParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpRtxParameters',
    },
    RTCRtpSendParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpSendParameters',
    },
    RTCRtpSynchronizationSource: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpSynchronizationSource',
    },
    RTCRtpTransceiverInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpTransceiverInit',
    },
    RTCRtpUnhandled: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCRtpUnhandled',
    },
    RTCSessionDescriptionInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCSessionDescriptionInit',
    },
    RTCSrtpKeyParam: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCSrtpKeyParam',
    },
    RTCSrtpSdesParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCSrtpSdesParameters',
    },
    RTCSsrcRange: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCSsrcRange',
    },
    RTCStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCStats',
    },
    RTCStatsEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCStatsEventInit',
    },
    RTCStatsReport: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'RTCStatsReport',
    },
    RTCTrackEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCTrackEventInit',
    },
    RTCTransportStats: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RTCTransportStats',
    },
    ReadableStreamReadDoneResult: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ReadableStreamReadDoneResult',
    },
    ReadableStreamReadValueResult: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ReadableStreamReadValueResult',
    },
    RegistrationOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RegistrationOptions',
    },
    RequestInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RequestInit',
    },
    ResponseInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ResponseInit',
    },
    RsaHashedImportParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaHashedImportParams',
    },
    RsaHashedKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaHashedKeyAlgorithm',
    },
    RsaHashedKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaHashedKeyGenParams',
    },
    RsaKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaKeyAlgorithm',
    },
    RsaKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaKeyGenParams',
    },
    RsaOaepParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaOaepParams',
    },
    RsaOtherPrimesInfo: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaOtherPrimesInfo',
    },
    RsaPssParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'RsaPssParams',
    },
    SVGBoundingBoxOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'SVGBoundingBoxOptions',
    },
    ScopedCredentialDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ScopedCredentialDescriptor',
    },
    ScopedCredentialOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ScopedCredentialOptions',
    },
    ScopedCredentialParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ScopedCredentialParameters',
    },
    ScrollIntoViewOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ScrollIntoViewOptions',
    },
    ScrollOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ScrollOptions',
    },
    ScrollToOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ScrollToOptions',
    },
    SecurityPolicyViolationEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'SecurityPolicyViolationEventInit',
    },
    ServiceWorkerMessageEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ServiceWorkerMessageEventInit',
    },
    ShadowRootInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ShadowRootInit',
    },
    ShareData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ShareData',
    },
    SpeechSynthesisErrorEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'SpeechSynthesisErrorEventInit',
    },
    SpeechSynthesisEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'SpeechSynthesisEventInit',
    },
    StaticRangeInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'StaticRangeInit',
    },
    StereoPannerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'StereoPannerOptions',
    },
    StorageEstimate: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'StorageEstimate',
    },
    StorageEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'StorageEventInit',
    },
    StoreExceptionsInformation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'StoreExceptionsInformation',
    },
    StoreSiteSpecificExceptionsInformation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'StoreSiteSpecificExceptionsInformation',
    },
    TextDecodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TextDecodeOptions',
    },
    TextDecoderOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TextDecoderOptions',
    },
    TextEncoderEncodeIntoResult: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TextEncoderEncodeIntoResult',
    },
    TouchEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TouchEventInit',
    },
    TouchInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TouchInit',
    },
    TrackEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TrackEventInit',
    },
    Transformer: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Transformer',
    },
    TransitionEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'TransitionEventInit',
    },
    UIEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'UIEventInit',
    },
    ULongRange: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ULongRange',
    },
    UnderlyingByteSource: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'UnderlyingByteSource',
    },
    UnderlyingSink: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'UnderlyingSink',
    },
    UnderlyingSource: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'UnderlyingSource',
    },
    VRDisplayEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'VRDisplayEventInit',
    },
    VRLayer: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'VRLayer',
    },
    VRStageParameters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'VRStageParameters',
    },
    WaveShaperOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WaveShaperOptions',
    },
    WebAuthnExtensions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WebAuthnExtensions',
    },
    WebGLContextAttributes: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WebGLContextAttributes',
    },
    WebGLContextEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WebGLContextEventInit',
    },
    WheelEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WheelEventInit',
    },
    WorkerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WorkerOptions',
    },
    WorkletOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'WorkletOptions',
    },
    EventListener: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventListener',
    },
    XPathNSResolver: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'XPathNSResolver',
    },
    ANGLE_instanced_arrays: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ANGLE_instanced_arrays',
    },
    AbortController: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AbortController',
    },
    AbortSignalEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AbortSignalEventMap',
    },
    AbortSignal: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AbortSignal',
    },
    AbstractRange: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AbstractRange',
    },
    AbstractWorkerEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AbstractWorkerEventMap',
    },
    AbstractWorker: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AbstractWorker',
    },
    AesCfbParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesCfbParams',
    },
    AesCmacParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesCmacParams',
    },
    AnalyserNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AnalyserNode',
    },
    Animatable: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Animatable',
    },
    AnimationEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnimationEventMap',
    },
    Animation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Animation',
    },
    AnimationEffect: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AnimationEffect',
    },
    AnimationEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AnimationEvent',
    },
    AnimationFrameProvider: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnimationFrameProvider',
    },
    AnimationPlaybackEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AnimationPlaybackEvent',
    },
    AnimationTimeline: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AnimationTimeline',
    },
    ApplicationCacheEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ApplicationCacheEventMap',
    },
    ApplicationCache: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ApplicationCache',
    },
    Attr: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Attr',
    },
    AudioBuffer: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioBuffer',
    },
    AudioBufferSourceNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioBufferSourceNode',
    },
    AudioContext: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioContext',
    },
    AudioDestinationNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioDestinationNode',
    },
    AudioListener: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioListener',
    },
    AudioNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioNode',
    },
    AudioParam: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioParam',
    },
    AudioParamMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioParamMap',
    },
    AudioProcessingEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioProcessingEvent',
    },
    AudioScheduledSourceNodeEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioScheduledSourceNodeEventMap',
    },
    AudioScheduledSourceNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioScheduledSourceNode',
    },
    AudioWorklet: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioWorklet',
    },
    AudioWorkletNodeEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioWorkletNodeEventMap',
    },
    AudioWorkletNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AudioWorkletNode',
    },
    AuthenticatorAssertionResponse: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AuthenticatorAssertionResponse',
    },
    AuthenticatorAttestationResponse: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AuthenticatorAttestationResponse',
    },
    AuthenticatorResponse: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'AuthenticatorResponse',
    },
    BarProp: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BarProp',
    },
    BaseAudioContextEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BaseAudioContextEventMap',
    },
    BaseAudioContext: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BaseAudioContext',
    },
    BeforeUnloadEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BeforeUnloadEvent',
    },
    BhxBrowser: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BhxBrowser',
    },
    BiquadFilterNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BiquadFilterNode',
    },
    Blob: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Blob',
    },
    Body: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Body',
    },
    BroadcastChannelEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BroadcastChannelEventMap',
    },
    BroadcastChannel: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BroadcastChannel',
    },
    ByteLengthQueuingStrategy: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ByteLengthQueuingStrategy',
    },
    CDATASection: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CDATASection',
    },
    CSSConditionRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSConditionRule',
    },
    CSSFontFaceRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSFontFaceRule',
    },
    CSSGroupingRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSGroupingRule',
    },
    CSSImportRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSImportRule',
    },
    CSSKeyframeRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSKeyframeRule',
    },
    CSSKeyframesRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSKeyframesRule',
    },
    CSSMediaRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSMediaRule',
    },
    CSSNamespaceRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSNamespaceRule',
    },
    CSSPageRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSPageRule',
    },
    CSSRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSRule',
    },
    CSSRuleList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSRuleList',
    },
    CSSStyleDeclaration: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSStyleDeclaration',
    },
    CSSStyleRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSStyleRule',
    },
    CSSStyleSheet: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSStyleSheet',
    },
    CSSSupportsRule: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CSSSupportsRule',
    },
    Cache: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Cache',
    },
    CacheStorage: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CacheStorage',
    },
    CanvasCompositing: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasCompositing',
    },
    CanvasDrawImage: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasDrawImage',
    },
    CanvasDrawPath: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasDrawPath',
    },
    CanvasFillStrokeStyles: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasFillStrokeStyles',
    },
    CanvasFilters: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasFilters',
    },
    CanvasGradient: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CanvasGradient',
    },
    CanvasImageData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasImageData',
    },
    CanvasImageSmoothing: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasImageSmoothing',
    },
    CanvasPath: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasPath',
    },
    CanvasPathDrawingStyles: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasPathDrawingStyles',
    },
    CanvasPattern: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CanvasPattern',
    },
    CanvasRect: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasRect',
    },
    CanvasRenderingContext2D: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CanvasRenderingContext2D',
    },
    CanvasShadowStyles: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasShadowStyles',
    },
    CanvasState: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasState',
    },
    CanvasText: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasText',
    },
    CanvasTextDrawingStyles: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasTextDrawingStyles',
    },
    CanvasTransform: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasTransform',
    },
    CanvasUserInterface: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasUserInterface',
    },
    CaretPosition: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CaretPosition',
    },
    ChannelMergerNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ChannelMergerNode',
    },
    ChannelSplitterNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ChannelSplitterNode',
    },
    CharacterData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CharacterData',
    },
    ChildNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ChildNode',
    },
    ClientRect: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ClientRect',
    },
    ClientRectList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ClientRectList',
    },
    Clipboard: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Clipboard',
    },
    ClipboardEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ClipboardEvent',
    },
    CloseEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CloseEvent',
    },
    Comment: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Comment',
    },
    CompositionEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CompositionEvent',
    },
    ConcatParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ConcatParams',
    },
    ConstantSourceNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ConstantSourceNode',
    },
    ConvolverNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ConvolverNode',
    },
    CountQueuingStrategy: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CountQueuingStrategy',
    },
    Credential: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Credential',
    },
    CredentialsContainer: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CredentialsContainer',
    },
    Crypto: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Crypto',
    },
    CryptoKey: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CryptoKey',
    },
    CryptoKeyPair: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CryptoKeyPair',
    },
    CustomElementRegistry: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CustomElementRegistry',
    },
    CustomEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'CustomEvent',
    },
    DOMError: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMError',
    },
    DOMException: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMException',
    },
    DOMImplementation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMImplementation',
    },
    DOML2DeprecatedColorProperty: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DOML2DeprecatedColorProperty',
    },
    DOMMatrix: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMMatrix',
    },
    SVGMatrix: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'SVGMatrix',
    },
    WebKitCSSMatrix: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'WebKitCSSMatrix',
    },
    DOMMatrixReadOnly: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMMatrixReadOnly',
    },
    DOMParser: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMParser',
    },
    DOMPoint: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMPoint',
    },
    SVGPoint: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'SVGPoint',
    },
    DOMPointReadOnly: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMPointReadOnly',
    },
    DOMQuad: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMQuad',
    },
    DOMRect: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMRect',
    },
    SVGRect: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'SVGRect',
    },
    DOMRectList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMRectList',
    },
    DOMRectReadOnly: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMRectReadOnly',
    },
    DOMSettableTokenList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMSettableTokenList',
    },
    DOMStringList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMStringList',
    },
    DOMStringMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMStringMap',
    },
    DOMTokenList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DOMTokenList',
    },
    DataCue: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DataCue',
    },
    DataTransfer: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DataTransfer',
    },
    DataTransferItem: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DataTransferItem',
    },
    DataTransferItemList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DataTransferItemList',
    },
    DeferredPermissionRequest: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DeferredPermissionRequest',
    },
    DelayNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DelayNode',
    },
    DeviceAcceleration: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DeviceAcceleration',
    },
    DeviceLightEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DeviceLightEvent',
    },
    DeviceMotionEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DeviceMotionEvent',
    },
    DeviceMotionEventAcceleration: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceMotionEventAcceleration',
    },
    DeviceMotionEventRotationRate: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DeviceMotionEventRotationRate',
    },
    DeviceOrientationEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DeviceOrientationEvent',
    },
    DeviceRotationRate: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DeviceRotationRate',
    },
    DhImportKeyParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DhImportKeyParams',
    },
    DhKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DhKeyAlgorithm',
    },
    DhKeyDeriveParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DhKeyDeriveParams',
    },
    DhKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DhKeyGenParams',
    },
    DocumentEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DocumentEventMap',
    },
    Document: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Document',
    },
    DocumentAndElementEventHandlersEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DocumentAndElementEventHandlersEventMap',
    },
    DocumentAndElementEventHandlers: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DocumentAndElementEventHandlers',
    },
    DocumentEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DocumentEvent',
    },
    DocumentFragment: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DocumentFragment',
    },
    DocumentOrShadowRoot: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DocumentOrShadowRoot',
    },
    DocumentTimeline: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DocumentTimeline',
    },
    DocumentType: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DocumentType',
    },
    DragEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DragEvent',
    },
    DynamicsCompressorNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'DynamicsCompressorNode',
    },
    EXT_blend_minmax: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EXT_blend_minmax',
    },
    EXT_frag_depth: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EXT_frag_depth',
    },
    EXT_sRGB: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EXT_sRGB',
    },
    EXT_shader_texture_lod: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EXT_shader_texture_lod',
    },
    EXT_texture_filter_anisotropic: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EXT_texture_filter_anisotropic',
    },
    ElementEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ElementEventMap',
    },
    Element: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Element',
    },
    ElementCSSInlineStyle: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ElementCSSInlineStyle',
    },
    ElementContentEditable: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ElementContentEditable',
    },
    ErrorEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ErrorEvent',
    },
    Event: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Event',
    },
    EventListenerObject: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventListenerObject',
    },
    EventSourceEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'EventSourceEventMap',
    },
    EventSource: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'EventSource',
    },
    EventTarget: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'EventTarget',
    },
    ExtensionScriptApis: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'ExtensionScriptApis',
    },
    External: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'External',
    },
    File: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'File',
    },
    FileList: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'FileList',
    },
    FileReaderEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'FileReaderEventMap',
    },
    FileReader: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'FileReader',
    },
    FocusEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'FocusEvent',
    },
    FocusNavigationEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'FocusNavigationEvent',
    },
    FormData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'FormData',
    },
    GainNode: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GainNode',
    },
    Gamepad: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Gamepad',
    },
    GamepadButton: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GamepadButton',
    },
    GamepadEvent: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GamepadEvent',
    },
    GamepadHapticActuator: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GamepadHapticActuator',
    },
    GamepadPose: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GamepadPose',
    },
    GenericTransformStream: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GenericTransformStream',
    },
    Geolocation: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Geolocation',
    },
    GeolocationCoordinates: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GeolocationCoordinates',
    },
    GeolocationPosition: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GeolocationPosition',
    },
    GeolocationPositionError: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'GeolocationPositionError',
    },
    GlobalEventHandlersEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GlobalEventHandlersEventMap',
    },
    GlobalEventHandlers: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'GlobalEventHandlers',
    },
    HTMLAllCollection: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLAllCollection',
    },
    HTMLAnchorElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLAnchorElement',
    },
    HTMLAppletElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLAppletElement',
    },
    HTMLAreaElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLAreaElement',
    },
    HTMLAudioElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLAudioElement',
    },
    HTMLBRElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLBRElement',
    },
    HTMLBaseElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLBaseElement',
    },
    HTMLBaseFontElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLBaseFontElement',
    },
    HTMLBodyElementEventMap: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HTMLBodyElementEventMap',
    },
    HTMLBodyElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLBodyElement',
    },
    HTMLButtonElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLButtonElement',
    },
    HTMLCanvasElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLCanvasElement',
    },
    HTMLCollectionBase: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HTMLCollectionBase',
    },
    HTMLCollection: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLCollection',
    },
    HTMLCollectionOf: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'HTMLCollectionOf',
    },
    HTMLDListElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLDListElement',
    },
    HTMLDataElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLDataElement',
    },
    HTMLDataListElement: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'HTMLDataListElement',
    },
    HTMLDetail
